server/handlers/shared_validators.js

```diff
@@ -23,3 +25,63 @@ exports.passwordBodyValidator = body("password")
   .isLength({ min: 6 })
   .trim()
   .escape();
+
+//products
+
+exports.productIdValidator = param("id").isMongoId();
+exports.farmerIdValidator = param("id").isMongoId();
+exports.productNameValidator = body("name")
+  .notEmpty()
+  .bail()
+  .isString()
+  .bail()
+  .isLength({ max: 25 })
+  .trim()
+  .escape();
+
+exports.productDescriptionValidator = body("description")
+  .notEmpty()
+  .bail()
+  .isString()
+  .bail()
+  .isLength({ max: 100 })
+  .trim()
+  .escape();
+
+exports.productCategoryValidator = body("category")
+  .if(body('category').exists())
+  .notEmpty()
+  .bail()
+  .isString()
+  .bail()
+  .isIn(Object.values(Product.Categories))
+  .trim()
+  .escape();
+
+exports.searchStringValidator = body("searchString")
+  .if(body('searchString').exists())
+  .notEmpty()
+  .bail()
+  .isString()
+  .bail()
+  .isLength({ max: 100 })
+  .trim()
+  .escape();
+
+exports.IDsValidator = body("IDs")
+  .if(body('IDs').exists())
+  .notEmpty()
+  .bail()
+  .isString()
+  .bail()
+  .isLength({ max: 100 })
+  .trim()
+  .custom(value => {
+    const splittedID = value.split(",");
+    if (splittedID.length == splittedID.filter(id => ObjectId.isValid(id)).length) {
+      return true
+    }
+    
+    return false;
```

Comment:
this is good, just a simplification in the return (the if is useless):
```suggestion
  return splittedIDs.length == splittedIDs.filter(id => ObjectId.isValid(id)).length;
```